src/main/java/com/google/sps/servlets/ProfileImage.java

```diff
@@ -52,8 +54,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       InputStream fileInputStream = filePart.getInputStream();
 
       UserImage.uploadObject("theglobetrotter-step-2020", fileName, fileInputStream);
-    } catch (Exception e) {
-      // TODO(#14): Catch more specific exceptions.
+    } catch (EntityNotFoundException e) {
+      response.getWriter().println("<p>Error getting user from datastore!</p>");
```

Comment:
This will not be seen by the user if the redirect still happens further down in the code.

Also (probably for a different PR), the details of the exception should be logged somewhere (perhaps the cloud logging service) so that the people who operate the application can monitor whether it is healthy and working as expected.

Reply:
Added TODO for that